db/migrate/20220716194537_merge_users.rb

```diff
@@ -0,0 +1,34 @@
+class MergeUsers < ActiveRecord::Migration[7.0]
+  def change
+
+    klass = Class.new(Partners::Base) do
+      self.table_name = 'partner_users'
+    end
+
+    ::User.reset_column_information
+
+    # ignore password issues for now
+    User.define_method(:password_required?) { false }
+
+    klass.all.each do |user|
+      begin
+        main_user = ::User.unscoped.find_by(email: user.email)
+        if main_user.nil?
+          attrs = user.attributes.except('id').merge(discarded_at: nil)
+          # null constraint on name for User table doesn't exist for Partner User table
+          attrs['name'] ||= "CHANGEME"
```

Comment:
```suggestion
          attrs['name'] ||= "CHANGEME"
```
Could we guess the name using the email?